src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -410,10 +413,14 @@ def module_is_offloaded(module):
         pipeline_is_sequentially_offloaded = any(
             module_is_sequentially_offloaded(module) for _, module in self.components.items()
         )
+        pipeline_has_bnb = any(
+            (_check_bnb_status(module)[1] or _check_bnb_status(module)[-1]) for _, module in self.components.items()
+        )
         if pipeline_is_sequentially_offloaded and device and torch.device(device).type == "cuda":
-            raise ValueError(
-                "It seems like you have activated sequential model offloading by calling `enable_sequential_cpu_offload`, but are now attempting to move the pipeline to GPU. This is not compatible with offloading. Please, move your pipeline `.to('cpu')` or consider removing the move altogether if you use sequential offloading."
-            )
+            if not pipeline_has_bnb:
+                raise ValueError(
+                    "It seems like you have activated sequential model offloading by calling `enable_sequential_cpu_offload`, but are now attempting to move the pipeline to GPU. This is not compatible with offloading. Please, move your pipeline `.to('cpu')` or consider removing the move altogether if you use sequential offloading."
+                )
```

Comment:
I think that here will still be a potential issue even with that. The issue is that hooks are added and might cause issues if we really move the model. To fix this, we should probably remove them. They were added [here](https://github.com/huggingface/accelerate/blob/bf4572b6ce0a534a9d73537485a0edf1d68144b8/src/accelerate/big_modeling.py#L355) in accelerate. We should just relax the case with 4-bit bnb by checking the version of bnb + removing the `getattr(model, "is_quantized", False)` condition. LMK if this is clear !

Reply:
Do you maybe want to push your changes directly to this PR?

Reply:
To fix this at the accelerate level !

Reply:
Alright. I will wait for it then

Reply:
please try with that ! https://github.com/huggingface/accelerate/pull/3223